.github/workflows/cleanup-cache.yml

```diff
@@ -0,0 +1,52 @@
+# Copied from https://github.com/actions/cache/blob/main/tips-and-workarounds.md#force-deletion-of-caches-overriding-default-cache-eviction-policy
+name: Cleanup Caches for Merged PRs
+
+on:
+  pull_request:
+    types: [closed, enqueued]
```

Comment:
tbh I'm not certain, I assume we still run CI for PRs in the merge queue and that's why it takes so long? in which case does it make sense to clean up the cache when enqueuing a PR?

Reply:
it's just how GHA caches work, that the merge queue cache is derived from the target branch, and is separate to that of the branch being merged